atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -288,52 +281,18 @@ protected CassandraKeyValueServiceImpl(Logger log,
         this.cassandraTables = new CassandraTables(clientPool, configManager);
     }
 
-    private LockLeader whoIsTheLockCreator() {
-        return leaderConfig
-                .map((config) -> config.whoIsTheLockLeader())
-                .orElse(LockLeader.I_AM_THE_LOCK_LEADER);
-    }
-
-    @Override
-    public synchronized void cleanUpOnInitFailure() {
-        // no-op
+    public boolean isInitialized() {
```

Comment:
It feels like there might still be value in having a small interface called `AsyncInitializable` (similar to `Closeable`), just to denote those classes that can by async-initialised. It would just have the two methods `isInitialized` and `initialize(bool)`.

Reply:
...maybe only if such classes have use for these methods? _\<continues reading\>_

Reply:
...looks like it's only `C*KVSImpl` that needs these two methods, so I'd say it's fine to leave as-is. Could be something to consider, though, if more classes with `InitializingWrapper`s start to need these options.